frontend/src/features/idir/search/api/idirSearch.ts

```diff
@@ -39,3 +40,22 @@ export const getDataBySearch = (
   searchURL.searchParams.set("take", take.toString());
   return httpGETRequest(searchURL.toString()).then((response) => response.data);
 };
+
+/**
+ * Searches the data with options and value entered by the user.
+ * @param SearchFields The search parameters.
+ * @returns The response from the API.
+ */
+export const getCompanyDataBySearch = (
+  { searchEntity, searchByFilter, searchString }: SearchFields,
+  { page = 0, take = 10 }: PaginationOptions,
+): Promise<PaginatedResponse<CompanyProfile>> => {
+  const searchURL = new URL(getSearchURLbyEntity(searchEntity));
+  searchURL.searchParams.set("searchColumn", searchByFilter);
```

Comment:
`legalName` and `clientNumber` params here right?

Reply:
just spotted that too - thanks!